lib/room_reservation.rb

```diff
@@ -0,0 +1,165 @@
+require "date"
+require_relative "calendar"
+require_relative "reservation"
+require_relative "block"
+require_relative "full_room"
+
+module Hotel
+  class RoomReservation
```

Comment:
This class seems like it would be better named something like `ReservationManager`.  Something that describes it's function.